examples/fl-tabular/federated.py

```diff
@@ -0,0 +1,186 @@
+
+
+class IncomeClassifier(nn.Module):
+    def __init__(self):
+        super(IncomeClassifier, self).__init__()
+        self.layer1 = None
+        self.layer2 = nn.Linear(128, 64)
+        self.output = nn.Linear(64, 1)
+        self.relu = nn.ReLU()
+        self.sigmoid = nn.Sigmoid()
+        self.dropout = nn.Dropout(0.5)
+
+    def forward(self, x):
+        if self.layer1 is None:
+            self.initialize_model(x.size(1))  # Initialize model with input dimension
+        x = self.relu(self.layer1(x))
+        x = self.dropout(x)
+        x = self.relu(self.layer2(x))
+        x = self.dropout(x)
+        x = self.sigmoid(self.output(x))
+        return x
+
+    def initialize_model(self, input_dim):
+        self.layer1 = nn.Linear(input_dim, 128)
+
+
+def train(model, train_loader, num_epochs=1):
+    criterion = nn.BCELoss()
+    optimizer = optim.Adam(model.parameters(), lr=0.0001)
+    model.train()
+    for epoch in range(num_epochs):
+        for X_batch, y_batch in train_loader:
+            optimizer.zero_grad()
+            outputs = model(X_batch)
+            loss = criterion(outputs, y_batch)
+            loss.backward()
+            optimizer.step()
+        print(f"Epoch [{epoch+1}/{num_epochs}], Loss: {loss.item():.4f}")
+
+
+def evaluate(model, test_loader):
+    model.eval()
+    criterion = nn.BCELoss()
+    loss = 0.0
+    correct = 0
+    total = 0
+    with torch.no_grad():
+        for X_batch, y_batch in test_loader:
+            outputs = model(X_batch)
+            batch_loss = criterion(outputs, y_batch)
+            loss += batch_loss.item()
+            predicted = (outputs > 0.5).float()
+            total += y_batch.size(0)
+            correct += (predicted == y_batch).sum().item()
+    accuracy = correct / total
+    loss = loss / len(test_loader)
+    print(f"Accuracy: {accuracy:.2f}")
+    return loss, accuracy
+
+
+def set_weights(net, parameters):
+    params_dict = zip(net.state_dict().keys(), parameters)
+    state_dict = OrderedDict({k: torch.tensor(v) for k, v in params_dict})
+    net.load_state_dict(state_dict, strict=True)
+
+
+def get_weights(net):
+    ndarrays = [val.cpu().numpy() for _, val in net.state_dict().items()]
+    return ndarrays
+
+
+class FlowerClient(NumPyClient):
+    def __init__(self, net, trainloader, testloader):
+        self.net = net
+        self.trainloader = trainloader
+        self.testloader = testloader
+
+    def fit(self, parameters, config):
+        set_weights(self.net, parameters)
+        train(self.net, self.trainloader)
+        return get_weights(self.net), len(self.trainloader), {}
+
+    def evaluate(self, parameters, config):
+        set_weights(self.net, parameters)
+        loss, accuracy = evaluate(self.net, self.testloader)
+        return loss, len(self.testloader), {"accuracy": accuracy}
+
+
+def client_fn(cid: str) -> Client:
+    train_loader, test_loader, input_dim = load_data(partition_id=int(cid))
```

Comment:
Preferably we don't re-create the FDS object w/ each call to `client_fn()`. I think it's better to create the FDS object once, then pass it to the `client_fn` so partitions (previously created) can be loaded when calling this `client_fn` function. One way of doing this is how the `examples/simulation_pytorch` does it.